src/engine/enginebuffer.cpp

```diff
@@ -1140,8 +1140,8 @@ void EngineBuffer::process(CSAMPLE* pOutput, const int iBufferSize) {
     m_pScaleST->setSampleRate(m_sampleRate);
     m_pScaleRB->setSampleRate(m_sampleRate);
 
-    bool bTrackLoading = m_iTrackLoading.loadAcquire() != 0;
-    if (!bTrackLoading && m_pause.tryLock()) {
+    bool hasStableTrack = m_pTrackLoaded->toBool() && m_iTrackLoading.loadAcquire() == 0;
+    if (hasStableTrack && m_pause.tryLock()) {
```

Comment:
This might fix the issue, though my guess is that the problem stems from a memory access ordering issue. Have you tried refactoring `m_iTrackLoading` to a `std::atomic<bool>` and then use `memory_order::seq_cst` accesses everywhere?

Reply:
The issue fortunately is pretty clear. 
We have identified https://github.com/mixxxdj/mixxx/commit/dff90d2c295f95c4a368bac32060707d728136f2
as the first failing commit. 
I have introduced an extra engine callback where the track is already ejected. m_pTrackLoaded is false than. 
This is required to clear the effect buffers.

Calling processTrackLocked() without a track is obviously wrong. 
I have not found out where exactly the bad memory access happens. Valgrind reveals only the symptom inside Soundtouch.

Reply:
I see that makes sense.

Reply:
Are you sure that the CO is not subject to data-races?

Reply:
Yes, because m_pTrackLoaded and m_iTrackLoading are atomic and only changed under the m_pause lock.